app/models/ingest_job.rb

```diff
@@ -118,6 +124,38 @@ def poll_for_remote(skip_push: false)
     is_pushed
   end
 
+  # Determine if a file is ready to be ingested.  This mainly validates that other concurrent parses for the same study
+  # will not interfere with validation for this current run
+  #
+  # * *returns*
+  #   - (Boolean) => T/F if file can launch PAPI ingest job
+  def can_launch_ingest?
+    case self.study_file.file_type
+    when /Matrix/
+      # expression matrices currently cannot be ingested in parallel due to constraints around validating cell names
+      # this block ensures that all other matrices have all cell names ingested and at least one gene entry, which
+      # ensures the matrix has validated
+      other_matrix_files = StudyFile.where(study_id: self.study.id, file_type: /Matrix/, :id.ne => self.study_file.id)
+      # only check other matrix files of the same type, as this is what will be checked when validating
+      similar_matrix_files = other_matrix_files.select {|matrix| matrix.is_raw_counts_file? == self.study_file.is_raw_counts_file?}
```

Comment:
I like that you did this part of the check in ruby instead of in the mongo query -- mongo nested equality/inequality is finicky